demo/screens/ListLandsScreen.js

```diff
@@ -1,60 +1,72 @@
 import React from "react";
-import { ListView } from "react-native";
 import ListLands from "@presentational/ListLands";
 import LandRow from "@presentational/LandRow";
+import { Action } from "tasit-sdk";
+import { abi as estateABI } from "../abi/EstateRegistry.json";
+import { abi as markplaceABI } from "../abi/Marketplace.json";
+const { Contract } = Action;
 
-const rows = [
-  {
-    id: 1,
-    name: "DRAGON KINGDOM",
-    img: require("../assets/images/lands/land1.png"),
-    priceMana: 13999,
-    priceUsd: 699,
-  },
-  {
-    id: 2,
-    name: "Parcel",
-    img: require("../assets/images/lands/land2.png"),
-    priceMana: 27985,
-    priceUsd: 1399,
-  },
-  {
-    id: 3,
-    name: "Golden Area",
-    img: require("../assets/images/lands/land3.png"),
-    priceMana: 19000,
-    priceUsd: 950,
-  },
-  {
-    id: 4,
-    name: "2 parcels away from Main Road to Genesis Plaza",
-    img: require("../assets/images/lands/land4.png"),
-    priceMana: 150000,
-    priceUsd: 7500,
-  },
-];
-
-const rowHasChanged = (r1, r2) => r1.id !== r2.id;
-
-const ds = new ListView.DataSource({ rowHasChanged });
+const estateAddress = "0x41b598a2c618b59b74540ac3afffb32f7971b37a";
+const marketplaceAddress = "0x07c0e972064e5c05f7b3596d81de1afd35457eae";
 
 export default class ListLandsScreen extends React.Component {
   state = {
-    dataSource: ds.cloneWithRows(rows),
+    rows: [],
   };
 
-  renderRow = land => {
+  async componentDidMount() {
+    const rows = await this.getSellOrders();
+    this.setState({ rows });
+  }
+
+  async getSellOrders() {
+    const orders = [];
+    const estateContract = new Contract(estateAddress, estateABI);
+    const totalSupply = await estateContract.totalSupply();
+
+    for (let estateId = 1; estateId <= Number(totalSupply); estateId++)
+      orders.push(this.getSellOrder(estateId));
+
+    return await Promise.all(orders);
```

Comment:
Are we assuming here that most estates will have a sell order?

Reply:
For the current local blockchain data, yes.

Reply:
Related to: https://github.com/tasitlabs/tasit/pull/79#discussion_r254306497

Reply:
Yep, we can resolve it based on what you said in the other thread.